packages/cloud_functions/cloud_functions_platform_interface/lib/src/platform_interface/platform_interface_firebase_functions.dart

```diff
@@ -19,55 +18,50 @@ import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 /// `implements` this interface will be broken by newly added
 /// [FirebaseFunctionsPlatform] methods.
 abstract class FirebaseFunctionsPlatform extends PlatformInterface {
-  static final Object _token = Object();
-
-  /// Create an instance using [app] and [region]
+  /// Create an instance using [app] and [region].
   FirebaseFunctionsPlatform(this.app, this.region) : super(token: _token);
 
-  static FirebaseFunctionsPlatform _instance;
+  /// Create an instance using [app] using the existing implementation
+  factory FirebaseFunctionsPlatform.instanceFor(
+      {FirebaseApp? app, String? region}) {
+    return FirebaseFunctionsPlatform.instance
+        .delegateFor(app: app, region: region);
+  }
+
+  static final Object _token = Object();
+
+  static FirebaseFunctionsPlatform? _instance;
 
   /// The [FirebaseApp] this instance was initialized with
-  final FirebaseApp app;
+  final FirebaseApp? app;
```

Comment:
`app` shouldn't be nullable I think. The public package should always send a FirebaseApp down with `instanceFor`. If the user does `instanceFor` without an `app` argument, then it uses `Firebase.app()`.

Reply:
Default instance for method channels has no args; 
![image](https://user-images.githubusercontent.com/5347038/105015454-7eaf8580-5a39-11eb-9135-39426d8be17e.png)